ddl/table.go

```diff
@@ -48,13 +48,9 @@ func (d *ddl) onCreateTable(t *meta.Meta, job *model.Job) error {
 	// Check the table.
 	for _, tbl := range tables {
 		if tbl.Name.L == tbInfo.Name.L {
-			if tbl.ID != tbInfo.ID {
```

Comment:
If `tbl.Name.L == tbInfo.Name.L` is true,  the job need to cancel. You can see the comments.